cg/meta/deliver_ticket.py

```diff
@@ -135,48 +88,15 @@ def concatenate(self, ticket: str, dry_run: bool) -> None:
                 continue
             if not dir_path.is_dir():
                 continue
-            for read_direction in [1, 2]:
-                same_direction: list[Path] = self.get_current_read_direction(
-                    dir_path=dir_path, read_direction=read_direction
-                )
-                total_size: int = self.get_total_size(files=same_direction)
-                output: Path = self.generate_output_filename(
-                    date=date, dir_path=dir_path, read_direction=read_direction
-                )
-                if dry_run:
-                    for file in same_direction:
-                        LOG.info(f"Dry run activated, {file} will not be appended to {output}")
-                else:
-                    LOG.info(f"Concatenating sample: {dir_path.name}")
-                    self.concatenate_same_read_direction(reads=same_direction, output=output)
-                if dry_run:
-                    continue
-                concatenated_size = output.stat().st_size
-                if total_size != concatenated_size:
-                    raise CgError("WARNING data lost in concatenation")
-
-                LOG.info(
-                    "QC PASSED: Total size for files used in concatenation match the size of the concatenated file"
-                )
-                self.remove_files(reads=same_direction)
-
-    def get_app_tag(self, samples: list) -> str:
-        app_tag = samples[0].application_version.application.tag
-        return app_tag
-
-    def check_if_concatenation_is_needed(self, ticket: str) -> bool:
```

Comment:
Why are we removing this check? Can't we just move it to the FastqFileService?

And I think it will be also a good idea to replace the one we have in clinical-delivery with this one based on tags, what do you think?
https://github.com/Clinical-Genomics/cg/blob/20b14ca20c459c038645d6999e42c7bb19f6bcde/cg/meta/deliver.py#L217-L218

Reply:
I don't  think we should move that logic into the `FastqFileService` concatenation method. It is a violation of the single responsible principle and creates a tight coupling between the service and the microbial workflows

Reply:
I see. If we end up using this module for FASTQ concatenation before starting the analysis (and replacing FastqHandler), then we won't be able to reuse it.

Reply:
But the question that remains is: why are we removing this check? Is it because this is a manual step that's only performed for specific pipelines that will always require concatenation?

Reply:
Me and Vadym discussed this and it can be removed since we are currently doing double concatenation when executing the `cg deliver ticket` command

Reply:
We've clarified that this check is duplicated since it calls the delivery API instead of the delivery ticket, as I initially thought. Resolving!